--- a/google-devices/shusky/display/panel-google-bigsurf.c
+++ b/google-devices/shusky/display/panel-google-bigsurf.c
@@ -206,8 +206,6 @@ static const struct exynos_dsi_cmd bigsurf_init_cmds[] = {
 				 0x00, 0x00, 0x00, 0x00, 0x60, 0x10, 0x70, 0x40,
 				 0x60, 0x10, 0x70, 0x40, 0x60, 0x10, 0x70, 0x40,
 				 0x60, 0x10, 0x70, 0x40),
-	/* Disable AOD power saving */
-	EXYNOS_DSI_CMD_SEQ(0xC7, 0x00),
 
 	/* CMD2, Page4 */
 	/* b/343326731: Extend DBI Flash Data Update Cycle time */
@@ -279,6 +277,27 @@ static const struct exynos_binned_lp bigsurf_binned_lp[] = {
 			      BIGSURF_TE2_RISING_EDGE_OFFSET, BIGSURF_TE2_FALLING_EDGE_OFFSET)
 };
 
+u8 freq_cmd[4] = {0x00, 0x43, 0x43, 0x03};
+module_param_array(freq_cmd, byte, NULL, 0644);
+
+static void bigsurf_send_dimming_freq_cmd(struct exynos_panel *ctx, int need_unlock, const u8 *cmd)
+{
+	if (need_unlock)
+		EXYNOS_DCS_BUF_ADD_SET(ctx, unlock_cmd_f0);
+
+	EXYNOS_DCS_BUF_ADD(ctx, 0xBD, 0x21, cmd[0], cmd[1], cmd[2], cmd[3]);
+
+	if (need_unlock) {
+		EXYNOS_DCS_BUF_ADD_SET(ctx, freq_update);
+		EXYNOS_DCS_BUF_ADD_SET_AND_FLUSH(ctx, lock_cmd_f0);
+	}
+}
+
+static void bigsurf_set_override_dimming(struct exynos_panel *ctx, int need_unlock)
+{
+	bigsurf_send_dimming_freq_cmd(ctx, need_unlock, freq_cmd);
+}
+
 static inline bool is_in_comp_range(int temp)
 {
 	return (temp >= 10 && temp <= 49);
@@ -530,6 +549,9 @@ static void bigsurf_set_panel_feat(struct exynos_panel *ctx, const u32 vrefresh,
 		else
 			EXYNOS_DCS_BUF_ADD(ctx, 0xBD, 0x21, 0x81, 0x83, 0x03, 0x03);
 	}
+	if (is_panel_enabled(ctx))
+		bigsurf_set_override_dimming(ctx, 0);
+
 	EXYNOS_DCS_BUF_ADD(ctx, 0xB0, 0x00, 0x10, 0xBD);
 	val = test_bit(FEAT_EARLY_EXIT, feat) ? 0x22 : 0x00;
 	EXYNOS_DCS_BUF_ADD(ctx, 0xBD, val);
@@ -1278,7 +1300,7 @@ const struct exynos_panel_desc google_bigsurf = {
 	/* supported HDR format bitmask : 1(DOLBY_VISION), 2(HDR10), 3(HLG) */
 	.hdr_formats = BIT(2) | BIT(3),
 	.max_luminance = 10000000,
-	.max_avg_luminance = 10000000,
+	.max_avg_luminance = 1200000,
 	.min_luminance = 5,
 	.modes = bigsurf_modes,
 	.num_modes = ARRAY_SIZE(bigsurf_modes),
